### ISSUE TITLE
LRScheduler ``.step()`` location inaccurate

### ISSUE THREAD

**Issue by thibaultdvx** [aramis-lab/clinicadl#805]
https://github.com/aramis-lab/clinicadl/blob/83d753494afd0fd33adc8683c6012eefcf482983/clinicadl/callbacks/factory/lr_scheduler.py#L163

Some LRSchedulers expects the ``scheduler.step()`` after every optimization steps (e.g. [CyclicLR](https://docs.pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CyclicLR.html#cycliclr)), whereas others expect it at the end of the epoch (e.g. [StepLR](https://docs.pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html#steplr)).
